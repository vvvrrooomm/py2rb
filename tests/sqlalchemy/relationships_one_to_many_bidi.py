from flask_login import current_user
from sqlalchemy import Column, ForeignKey, Integer, String, Boolean
from sqlalchemy.orm import relationship
from flask_sqlalchemy import SQLAlchemy

db = SQLAlchemy()


class Parent(Base):
    __tablename__ = "parent_table"
    id = Column(Integer, primary_key=True)
    children = relationship("Child", back_populates="parent")


class Child(Base):
    __tablename__ = "child_table"
    id = Column(Integer, primary_key=True)
    parent_id = Column(Integer, ForeignKey("parent_table.id"))
    parent = relationship("Parent", back_populates="children")