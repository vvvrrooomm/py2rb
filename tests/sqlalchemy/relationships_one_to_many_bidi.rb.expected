# frozen_string_literal: true

require 'module'

using PythonPrintEx
using PythonStringCountEx

db = SQLAlchemy()
class Parent < ApplicationRecord
  #id = Column(Integer, primary_key: true)
  belongs_to :child #
end
class Child < ApplicationRecord
  #id = Column(Integer, primary_key: true)
  #parent_id = Column(Integer, ForeignKey("parent_table.id"))
  has_many :parents #
end
